packages/library-base/src/commonMain/kotlin/io/realm/Realm.kt

```diff
@@ -126,3 +146,13 @@ interface Realm : TypedRealm {
      */
     fun close()
 }
+
+/**
+ * Returns a [RealmQuery] matching the predicate represented by [query].
+ *
+ * Reified convenience wrapper for [Realm.query].
+ */
+inline fun <reified T : RealmObject> Realm.query(
```

Comment:
I wouldn't expect there to be many (if any) use cases for the non-reified versions. So we could rename that and/or make it internal. Then I think the IDE will automatically import the extension method. It is just really annoying to manually add the import.

Reply:
I'm a bit torn on this....IMO the reified version is easier on the eye when reading code, but at least I find it harder to type in the IDE because the autocomplete is less helpful.

Having two different names for the same thing would be super annoying, but only having extension functions would also make the API less discoverable :(

Reply:
If we don't consider the extension function discoverable if it is the only option, how would people ever find it if we keep it but the IDE always defaults to the non-reified version? 

IMO this ends up being worthless if people need to add the import themselves. I don't know if it too drastic to make the inner method internal, but while having two names is really annoying, it is better than worthless. 

But we can leave it as is and then monitor the frustrations.

Reply:
I should probably play around a little more with this in the IDE to get a more informed opinion, but right now I would be leaning towards:

1) Use the same name for both methods (mostly since I don't have any ideas on how to name them).
2) Reference the reified version from Dokka.
3) Always use the reified version in samples.
4) Find and upvote or create an issue for the issue with finding and importing the reified version.

Reply:
Just a not on the usage of reified functions, not necessarily linked to this one in particular: when doing exhaustive testing where we need to iterate through different properties it may not be possible to use the reified version. But I'm all onboard with using it in the samples.